src/components/MAPI/apps/utils.ts

```diff
@@ -268,3 +268,209 @@ export function filterClusters(
     }
   });
 }
+
+export async function ensureConfigMapForApp(
+  client: IHttpClient,
+  auth: IOAuth2Provider,
+  namespace: string,
+  appName: string,
+  contents: string
+) {
+  const userConfigMapName = getUserConfigMapName(appName);
+  const configMap = await ensureAppUserConfigMap(
+    client,
+    auth,
+    namespace,
+    userConfigMapName,
+    contents
+  );
+
+  if (!configMap) return null;
```

Comment:
When `!configMap` is true? what happens when `null` is returned?

Reply:
In this case, `configMap` can be either `corev1.IConfigMap` or `null`, so `!configMap` is `true` when `configMap` is `null`.

`null` is returned when creating a `ConfigMap` resource is not necessary: If the given contents is an empty string and an existing `ConfigMap` does not exist, then we skip creating one.

If an empty string is supplied but a `ConfigMap` resource already exists, then we'll update it to have empty contents, and we won't return `null`